src/integrations/feed/types.ts

```diff
@@ -0,0 +1,66 @@
+import { Context } from '../../Context';
+
+export type FeedResponse = [postId: string, metadata: string | undefined][];
+
+export type FeedProvider = {
+  enable?: boolean;
+  remove_engaged_posts?: boolean;
+  page_size_fraction: number;
+  penalise_source?: boolean;
+  penalise_source_seed?: number;
+  fallback_provider?: string;
+};
+
+export type PersonalizedProvider = FeedProvider & {
+  content_factor: {
+    tag_rank?: boolean;
+    source_rank?: boolean;
+    content_curation_rank?: boolean;
+  };
+};
+
+export type VectorSimilarityProvider = FeedProvider & {
+  disable_vector_search_cache?: boolean;
+};
+
+export type FeedConfig = {
+  feed_id: string;
+  user_id?: string;
+  feed_config_name?: string;
+  page_size: number;
+  offset: number;
+  total_pages: number;
+  fresh_page_size?: string;
+  allowed_tags?: string[];
+  blocked_tags?: string[];
+  blocked_sources?: string[];
+  squad_ids?: string[];
+  providers?: Record<string, FeedProvider>;
+};
+
+export interface FeedConfigGenerator {
+  generate(
+    ctx: Context,
+    userId: string | undefined,
```

Comment:
From the `Context`, we should be able to fetch the user's id right?

Reply:
So some feeds may ask for a different userId or not provide a userId at all even if you are logged in. So I wanted to be explicit about it